### PR TITLE
Added support for emoji

### PR DESCRIPTION
**Pull request by gpeal**

### CODE REVIEW

File: lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java [hunk 1]
```diff
@@ -30,7 +32,8 @@
 import java.util.Map;
 
 public class TextLayer extends BaseLayer {
-  private final char[] tempCharArray = new char[1];
+  // Capacity is 2 because emojis are 2 characters.
+  private final StringBuilder stringBuilder = new StringBuilder(2);
```
Comment (digitalbuddha) [inline lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: I'm a bit hazy on this, are emoji's always 2 characters?

File: lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java [hunk 2]
```diff
@@ -359,6 +363,17 @@ private void drawCharacter(char[] character, Paint paint, Canvas canvas) {
     return contents;
   }
 
+  private String codePointToString(int codePoint) {
```
Comment (digitalbuddha) [inline lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: move to a utility? Doesn't seem specific to `TextLayer`

File: lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java [hunk 3]
```diff
@@ -7,7 +7,9 @@
 import android.graphics.Path;
 import android.graphics.RectF;
 import android.graphics.Typeface;
+import android.util.LruCache;
```
Comment (digitalbuddha) [inline lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: I see the inport for LruCache but I don't see any references to it in code.  Did you mean to remove the import when using `SparseArrayCompat` as your cache construct?

Comment (gpeal) [reply lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: Yup :)

File: lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java (hunk 2 above)
Comment (gpeal) [reply lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: @digitalbuddha It's not used anywhere else though.

Comment (digitalbuddha) [reply lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: fair enough, I know it currently does not have a test around it. If you were to write a test would you then create an impl of TextLayer to do it? Its a nitpick and your choice

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/c054bdc5b2e4361a295274b26ad8894ef43d77db-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/c054bdc5b2e4361a295274b26ad8894ef43d77db-android28)

File: lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java (hunk 2 above)
Comment (gpeal) [reply lottie/src/main/java/com/airbnb/lottie/model/layer/TextLayer.java]: I just use the snapshot tests as a test. We could do unit tests too but they would be redundant with the snapshots.

**Comment by gpeal** [comment]
**Snapshot Tests**
**28**: [Report](https://happo.io/a/27/report/7c51d2440a8cdbbd97ba57a1aed3e762d8e89cba-android28) [Diff](https://happo.io/a/27/p/27/compare/master-android28/7c51d2440a8cdbbd97ba57a1aed3e762d8e89cba-android28)

**Pull request merged**